### DISCUSSION TITLE
[vscode] Install dependencies automatically

### DISCUSSION THREAD

**Discussion by akss13** [microsoft/vcpkg#17826] [Ideas]
I'm working on c++ project on a linux machine with vscode and it uses some boost libraries. I've installed them on my system using vcpkg and build it using the toolchain provided by vcpkg. My question is: How do I define the dependencies so that they automatically install on a different system, if they were to build it? Conan has a way of doing it by defining the dependencies in conanfile.txt. How do I do the same with vcpkg?
I've found [autovcpkg](https://github.com/fungos/autovcpkg) which does the job I'm looking to do but can the same be done natively inside cmakelists.txt or by vcpkg itself?

**Comment by hros** [comment]
This question is over two years old. 
I am taking my first steps with `vcpkg`, and would like to know the recommended way of making a repository with vcpkg dependecies available to other developers.
I guess I could make a vcpkg package out of my repository, this is probably the recommended way for production ready public deployment
What about sharing development branches? What is recommended way to share all the vcpkg dependencies between developers? is it through the aforementioned [autovcpkg ](https://github.com/fungos/autovcpkg) tool?

**Comment by Osyotr** [comment]
@akss13 @hros you are looking for manifest mode + binary caching
https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md
https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md

**Comment by JackBoosY** [comment]
Yeah, the VS Code vcpkg plugin is on the way.

**Comment by hros** [comment]
great to hear that, @JackBoosY 
Is there a beta (or alpha) version I can try out?

**Reply by JackBoosY** [reply]
Temporary no.
